lab_4_retrieval_w_clustering/main.py

```diff
@@ -182,7 +256,28 @@ def retrieve_relevant_documents(self, query: str, n_neighbours: int) -> list[tup
         Returns:
             list[tuple[float, str]]: Relevant documents with their distances.
         """
-
+        if (n_neighbours > len(self._db.get_vectors()) or not isinstance(query, str)
+                or query is None or len(query) == 0 or not isinstance(n_neighbours, int)):
+            raise ValueError
+        if n_neighbours < 0:
+            raise ValueError
+        tokenized_query = self._tokenizer.tokenize(query)
+        if tokenized_query is None or len(tokenized_query) == 0:
+            raise ValueError
+        query_vector = self._vectorizer.vectorize(tokenized_query)
+        if query_vector is None or len(query_vector) == 0:
+            raise ValueError
+        vector_data = [x[1] for x in self._db.get_vectors()]
+        answer = self._calculate_knn(query_vector, vector_data, n_neighbours)
+        print(answer)
```

Comment:
rm

Reply:
What do you mean by "rm"? Remove, rename?